spec/book_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'rspec'
+require './book'
+require './rental'
+require './person'
+
+describe Book do
+  before(:each) do
+    @book = Book.new('if tomorrow comes', 'sidney sheldon')
+  end
+
+  context 'when creating a new book' do
+    it 'should create a new book' do
+      book = Book.new('if tomorrow comes', 'sidney sheldon')
+      expect(book.title).to eq('if tomorrow comes')
+      expect(book.author).to eq('sidney sheldon')
+    end
+  end
+
+  context 'when I call the add_rental method' do
+    it 'should add a book to the rentals' do
+      book = Book.new('The Lord of the Rings', 'J.R.R. Tolkien')
+      rental = double('rental')
+      expect(rental).to receive(:book=).with(book)
+      expect(book.add_rental(rental)).to eq(book)
+    end
+  end
```

Comment:
- [ ] I would suggest modifying this test so it can provide a lot more confidence that your code is working, at the moment, your test doesn't prove that the rental is getting saved. I suggest the following:

```suggestion
    it 'should add a book to the rentals' do
      book = Book.new("Title", "Author")
      person = Person.new(24, "Bob")
      rental = Rental.new("2023-04-27", book, person)
      
      book.add_rental(rental)
      
      expect(book.rentals).to eq [rental]
    end
  end
```

This way, you're making sure that the `Book#add_rental` method is actually adding the rental to the rentals array as expected! 🤓 🔧